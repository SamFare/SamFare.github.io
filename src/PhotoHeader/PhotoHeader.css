.photoHeader {
    width: 100%;
    height: 700px;
    position: relative;
    text-align: center;
    color: white;
    overflow: hidden;
    margin: -100px 0 0 0;
}

.bottomRight {
    position: absolute;
    bottom: 8px;
    right: 16px;
    font-size: 150px;
    padding: 0;
}

.photo {
    width: 100%;
    overflow: hidden;
}