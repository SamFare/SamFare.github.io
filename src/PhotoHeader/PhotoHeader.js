import React from "react";
import PropTypes from "prop-types";

import css from "./PhotoHeader.css";

export default class PhotoHeader extends React.Component{
  constructor(props) {
    super(props);
  }

  static get propTypes() {
    return {
      text: PropTypes.string, 
      imgUrl: PropTypes.string 
    };  
  }

  render(){
    return (
      <div className={css.photoHeader}>
        <img className={css.photo} src="assets/about-page-pic.jpg" />
        <div className={css.bottomRight}>{this.props.text}</div>
      </div> 
    );
  }
}