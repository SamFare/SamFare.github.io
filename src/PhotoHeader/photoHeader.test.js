import React from "react";
import { mount } from "enzyme";

import PhotoHeader from "./PhotoHeader.js";

test("The about page contains a photo background", () => { 
  const wrapper = mount(<PhotoHeader />);
  const photo = wrapper.find(".qa-photo");
  expect(photo.length).toEqual(1);
});

test.each`
fileUrl
${"./test.png"}
${"./test2.png"}
`("the page uses the image provided", ({fileUrl}) => { 
  const wrapper = mount(<PhotoHeader imgUrl={fileUrl}/>);
  expect(wrapper.find(".qa-photo").prop("style")).toHaveProperty("backgroundImage", fileUrl);
});

