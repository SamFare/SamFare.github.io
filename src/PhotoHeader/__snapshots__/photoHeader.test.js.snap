// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it has the image that was passed in 1`] = `
<div
  className="undefined qa-photo"
  style={
    Object {
      "backgroundImage": "url(testMob.png)",
    }
  }
>
  <noscript />
  <div>
    <noscript />
  </div>
</div>
`;
